ReactiveCocoa/Swift/Signal.swift

```diff
@@ -1026,8 +1026,8 @@ private struct ZipState<Value> {
 	var values: [Value] = []
 	var completed = false
```

Comment:
Maybe this should have be:

```swift
private struct ZipState<Left, Right> {
    var values: (left: [Left], right: [Right) = ([], [])
    var completed: (left: Bool, right: Bool) = (false, false)

    var isFinished: Bool {
        return values.left.isEmpty && values.left.completed || values.right.isEmpty && values.right.completed
    }
}
```

(It's too bad that Swift doesn't have a builtin `Pair` type!)